exchanges/kucoin/kucoin_websocket.go

```diff
@@ -941,16 +941,22 @@ func (ku *Kucoin) processOrderbook(respData []byte, symbol string) error {
 	return nil
 }
 
-func (ku *Kucoin) processMarketSnapshot(respData []byte, instrument string) error {
+func (ku *Kucoin) processMarketSnapshot(respData []byte) error {
 	response := WsSpotTicker{}
 	err := json.Unmarshal(respData, &response)
 	if err != nil {
 		return err
 	}
-	pair, err := currency.NewPairFromString(instrument)
+	pair, err := currency.NewPairFromString(response.Data.Symbol)
 	if err != nil {
 		return err
 	}
+	assetEnabledPairs := ku.listOfAssetsCurrencyPairEnabledFor(pair)
```

Comment:
can range over this map.

ALSO:
Suggestion but OUT OF SCOPE: Can update `listOfAssetsCurrencyPairEnabledFor` to only return a slice of assets which are associated with the enabled currency. Reduces a lot of what is going on below.

Reply:
Ranged over the map and amended the function to return a slice, but had to change a few functions due to that :sweat_smile: